pliers/extractors/audio.py

```diff
@@ -488,4 +494,101 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
     For details on argument specification visit:
     https://librosa.github.io/librosa/effect.html.'''
 
+
     _feature = 'percussive'
+
+
+class AudiosetLabelExtractor(AudioExtractor):
+
+    ''' Extract probability of audio event classes based on AudioSet-YouTube 
+    corpus using a YAMNet architecture. Code available at:
+    https://github.com/tensorflow/models/tree/master/research/audioset/yamnet 
+
+    Args:
+    hop_size (float): size of the window (in seconds) on which label extraction
+        is performed.
+    top_n (int): specifies how many of the highest label probabilities are 
+        returned. If not defined, returns probabilities for all labels.
+    label_subset (list): specifies subset of labels for which probabilities 
+        are to be returned. A comprehensive list of labels is available in the
+        audioset/yamnet repository (yamnet_class_map.csv).
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('hop_size', 'top_n', 'yamnet_kwargs')
+
+    def __init__(self, hop_size=0.1, top_n=None, label_subset=None,
+                 **yamnet_kwargs):
+        verify_dependencies(['yamnet'])
+        verify_dependencies(['tensorflow'])
+
+        MODULE_PATH = path.dirname(yamnet.__file__)
+        WEIGHTS_PATH = path.join(MODULE_PATH, 'yamnet.h5')
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+
+        self.hop_size = hop_size
+        self.yamnet_kwargs = yamnet_kwargs
+        self.params = yamnet.params
+        self.params.PATCH_HOP_SECONDS = hop_size
+        for par, v in self.yamnet_kwargs.items():
```

Comment:
The yamnet model expects params to be a module with attributes, rather than a dictionary. 
Leaning towards leaving it as it is instead of reconverting back to this format within _extract. Simplified the logic a bit by removing the conditional statement (adding extra stuff to params shouldn't matter, it would simply not be used).